app/controllers/pdl.py

```diff
@@ -1212,6 +1228,71 @@ def generate_title_i_total_state_distribution_response(title_id, start_year, end
 
     return result_dict
 
+def generate_title_ii_total_state_distribution_response(title_id, start_year, end_year):
+    session = Session()
+
+    # construct the query
+    program_query = (
+        session.query(
+            Payment.state_code.label('state'),
+            Payment.year.label('year'),
+            Payment.payment.label('totalPaymentInDollars')
+        ).filter(
+            Payment.title_id == title_id,
+            Payment.year.between(start_year, end_year),
+            # when calculating the title level summaries, only calculate Total CRP (sub_program_id == 102) from CRP (program_id == 108)
+            or_(
+                and_(
+                    Payment.program_id == 108,
+                    Payment.sub_program_id == 102
+                ),
+                Payment.program_id != 108
+            )
+        )
+    )
+
+    # execute the query
+    result = program_query.all()
+
+    # create a nested dictionary to store data by year and state
+    data_by_year_and_state = defaultdict(
+        lambda: defaultdict(lambda: {'totalPaymentInDollars': 0, 'totalRecipients': 0}))
+    all_years_summary = defaultdict(lambda: {'totalPaymentInDollars': 0, 'totalRecipients': 0})
+
+    for record in result:
+        state, year, payments = record
+        entry = data_by_year_and_state[year][state]
+        entry['state'] = state
+        entry['totalPaymentInDollars'] += payments
+
+        # add to all years summary
+        summary = all_years_summary[state]
+        summary['state'] = state
+        summary['totalPaymentInDollars'] += payments
+
+    # sort by total payment
+    sorted_data_by_year = {}
+    for year, states in data_by_year_and_state.items():
+        sorted_entries = sorted(states.values(), key=lambda x: x['totalPaymentInDollars'], reverse=True)
+        sorted_data_by_year[year] = sorted_entries
+
+    # round the total payment
+    for year, states in sorted_data_by_year.items():
+        for state in states:
+            state['totalPaymentInDollars'] = round(state['totalPaymentInDollars'], 2)
+
+    # all years summary
+    sorted_summary = sorted(all_years_summary.values(), key=lambda x: x['totalPaymentInDollars'], reverse=True)
+
+    # round the total payment
+    for state in sorted_summary:
+        state['totalPaymentInDollars'] = round(state['totalPaymentInDollars'], 2)
+
+    sorted_data_by_year['2014-2021'] = sorted_summary
```

Comment:
Please remove hard-coded years and use start and end-year variables instead. Thanks